freshmaker/models.py

```diff
@@ -617,6 +617,33 @@ def get_lowest_build_id(cls, session):
             return 0
         return build.build_id
 
+    @classmethod
+    def get_original_nvr_by_rebuilt_nvr(cls, session, rebuilt_nvr):
+        """
+        Search for an original build event, that has been built by FM, and
+            get original_nvr from it
+
+        :param session: database session
+        :param str rebuilt_nvr: rebuilt NVR to look build by
+        :rtype: str or None
+        :return: original NVR from the first FM build for given NVR
+        """
+        original_nvr = None
+        # build event should be only one in database, or raise an error
+        build_event = db.session.query(ArtifactBuild).filter(
+            ArtifactBuild.rebuilt_nvr == rebuilt_nvr,
+            ArtifactBuild.type == ArtifactType.IMAGE.value,
+        ).distinct().one_or_none()
```

Comment:
Could you please explain why `distinct` is required here?

Reply:
Oh yes, I have it because there was a join previously, I will remove it.